contributing.md

```diff
@@ -1,34 +1,95 @@
 # Contributing to FingerprintJS
 
+Thanks for taking the time to contribute!
+Here you can find ways to make FingerprintJS better, as well as tips and guidelines.
+
+This project and everyone participating in it is governed by the [Code of Conduct](code_of_conduct.md).
+By participating, you are expected to uphold this code.
+
+## How you can contribute
+
+### Reporting an issue
+
+If you've noticed a bug, have an idea or a question,
+feel free to [create an issue](https://github.com/fingerprintjs/fingerprintjs/issues/new/choose) or [start a discussion](https://github.com/fingerprintjs/fingerprintjs/discussions/new/choose).
+
+Before you start, please [search](https://github.com/search?q=repo%3Afingerprintjs%2Ffingerprintjs&type=code) for your topic.
+There is a chance it has already been discussed.
+
+When you create an issue, the description is pre-filled with a template text.
+Please fill the missing information carefully, it will help us solve your issue faster.
+If you want to show a code or a FingerprintJS output, wrap it in ` ``` ` blocks and don't truncate.
+
+### Creating a pull request
+
+If you want to fix a bug, add an entropy source or make another change to the code, you can [create a pull request](https://docs.github.com/en/get-started/exploring-projects-on-github/contributing-to-a-project).
+
+After you clone the repository, check the [Working with the code](#working-with-the-code) section to learn how to run, check and build the code.
+
+Please follow these guidelines so that we can accept your code changes:
+- Your code quality should be at least as good as the code that you modify;
+- Your code style (syntax, naming, coding patterns, etc) should follow the style of the other repository code;
+- The new code should be covered with automatic tests;
+- All the checks described in the [Working with the code](#working-with-the-code) section must pass successfully.
+    You may create a draft pull request in this repository to run the checks automatically by GitHub Actions,
+    but the tests won't run on BrowserStack until a FingerprintJS maintainer approves it;
+- If you want to add an entropy source (fingerprint component), follow the [How to make an entropy source](#how-to-make-an-entropy-source) instructions carefully;
+- The change shouldn't be breaking, that is FingerprintJS users don't need to change code on their side after your changes;
+- Don't add dependencies (such as Node packages) unless necessary;
+- Don't make changes not related to the PR goal, try to keep your changes as small as possible;
```

Comment:
..unrelated the stated pull request purpose. Please strive to introduce as few changes as possible.